### PR TITLE
RDK-27957 Bring up TextToSpeech plugin code from master to sprint/2009

### PR DESCRIPTION
**Pull request by nrajan002c**
Signed-off-by: nrajan002c <nambi_rajang@cable.comcast.com>

### CODE REVIEW

**Review by varumugam123 — APPROVED** [review]

File: TextToSpeech/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        impl/logger.cpp
+        )
+set_target_properties(${MODULE_NAME} PROPERTIES
+        CXX_STANDARD 11
+        CXX_STANDARD_REQUIRED YES)
+
+list(APPEND CMAKE_MODULE_PATH
+        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
+
+find_program(PKG_CONFIG_EXECUTABLE NAMES pkg-config DOC "pkg-config executable")
```
Comment (emutavchi) [inline TextToSpeech/CMakeLists.txt:22]: Why not create FindGstreamer.cmake or similar and use find_package? note, there is PkgConfig find package already.

File: TextToSpeech/TextToSpeech.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (emutavchi) [inline TextToSpeech/TextToSpeech.json:1]: Isn't this auto-generated? probably not needed to be checked in.

File: TextToSpeech/impl/TTSManager.cpp [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+TTSManager::TTSManager(TTSEventCallback *callback) :
+    m_callback(callback),
+    m_speaker(NULL),
+    m_ttsEnabled(false) {
+
+    TTSLOG_TRACE("TTSManager::TTSManager");
+
+    // Load configuration from file
+    loadConfigurationsFromFile(TTS_CONFIGURATION_FILE);
```
Comment (emutavchi) [inline TextToSpeech/impl/TTSManager.cpp:125]: This should not be allowed in PROD builds.

File: TextToSpeech/impl/TTSSpeaker.cpp [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
+    }
+
+    m_source = gst_element_factory_make("souphttpsrc", NULL);
+
+    // create soc specific elements
+#if defined(BCM_NEXUS)
+    GstElement *decodebin = NULL;
+    decodebin = gst_element_factory_make("brcmmp3decoder", NULL);
+    m_audioSink = gst_element_factory_make("brcmpcmsink", NULL);
+#endif
```
Comment (emutavchi) [inline TextToSpeech/impl/TTSSpeaker.cpp:371]: I think this is against VAP requirements, i.e. we shouldn't use SoC specific plugins directly. has to be addressed sooner or later

File: TextToSpeech/impl/TTSManager.cpp (hunk 3 above)
Comment (varumugam123) [reply TextToSpeech/impl/TTSManager.cpp:125]: Hi Eugene, could you let us know the concern?

File: TextToSpeech/impl/TTSSpeaker.cpp (hunk 4 above)
Comment (varumugam123) [reply TextToSpeech/impl/TTSSpeaker.cpp:371]: Hi Eugene, I have a doubt. Should rdk services as well follow VAP?

File: TextToSpeech/impl/TTSManager.cpp (hunk 3 above)
Comment (emutavchi) [reply TextToSpeech/impl/TTSManager.cpp:125]: This is the general rule. in nutshell, any dev configuration left on the box should not have any impact on prod image

File: TextToSpeech/impl/TTSSpeaker.cpp (hunk 4 above)
Comment (emutavchi) [reply TextToSpeech/impl/TTSSpeaker.cpp:371]: Good question, I think it makes sense for services as well. I mean it should just work in a generic way.

File: TextToSpeech/CMakeLists.txt (hunk 1 above)
Comment (nrajan002c) [reply TextToSpeech/CMakeLists.txt]: Modified the code to use FindGstreamer.cmake.

File: TextToSpeech/TextToSpeech.json (hunk 2 above)
Comment (nrajan002c) [reply TextToSpeech/TextToSpeech.json:1]: Removed the TextToSpeech.json.

File: TextToSpeech/impl/TTSManager.cpp (hunk 3 above)
Comment (nrajan002c) [reply TextToSpeech/impl/TTSManager.cpp:125]: We are planning to add the default configuration as a config setting in the ticket RDK-20988.

File: TextToSpeech/impl/TTSSpeaker.cpp (hunk 4 above)
Comment (nrajan002c) [reply TextToSpeech/impl/TTSSpeaker.cpp:371]: We will remove the explicit SOC specific plugins as part of RDK-20988.

**Review by emutavchi — APPROVED** [review]

**Pull request merged**